src/chains/ethereum/ethereum/src/helpers/storage-range-at.ts

```diff
@@ -0,0 +1,129 @@
+import { Data, findInsertPosition } from "@ganache/utils";
+import { decode } from "@ganache/rlp";
+
+import type { LevelUp } from "levelup";
+import type {
+  StorageRangeAtResult,
+  StorageRecords
+} from "@ganache/ethereum-utils";
+import { BaseTrie as Trie } from "merkle-patricia-tree";
+
+type TrieData = { key: Buffer; value: Buffer };
+type HashedKeysWithValues = {
+  length: number;
+  values: Buffer[];
+  keys: Buffer[];
+  nextKey: Data | null;
+};
+
+const compare = (a: Buffer, b: Buffer) => a.compare(b) < 0;
```

Comment:
I guess this is mostly just moving stuff around, but it seems unnecessary to provide a comparator, when we don't provide anything but the default buffer comparator. I'd rather just have this function hard coded within the `findInsertPosition` function.

Reply:
The only reason I'll push back on this a tiny bit is because `findInsertPosition` was ported from C++, and that implementation takes a comparator, and it can compare _anything_, not just Buffers. If we hardcode then comparator we'll have to update `findInsertPosition` to be monomorphic (Buffers only).

Reply:
Defs not worthwhile holding up this PR, but it might be worthwhile considering that especially if performance is important for this function. I _guess_ the comparator can be invoked a lot?